js/src/Debug/Widget/HTTPRequests.vue

```diff
@@ -0,0 +1,298 @@
+            });
+        }
+        // Sort by column
+        sorted.sort((a, b) => {
+            let a_val = a[sorted_col.value];
+            let b_val = b[sorted_col.value];
+            if (sorted_col.value === 'duration') {
+                a_val = parseFloat(a_val);
+                b_val = parseFloat(b_val);
+            }
+            if (a_val === b_val) {
+                return 0;
+            }
+            if (sort_dir.value === 'asc') {
+                return a_val < b_val ? -1 : 1;
+            } else {
+                return a_val > b_val ? -1 : 1;
+            }
+        });
+        return sorted;
+    });
+
+    function setSortedCol(col) {
+        if (sorted_col.value === col) {
+            if (sort_dir.value === 'asc') {
+                sort_dir.value = 'desc';
+            } else {
+                sort_dir.value = 'asc';
+            }
+        } else {
+            sorted_col.value = col;
+            sort_dir.value = 'asc';
+        }
+    }
+
+    const seen_request_ids = props.ajax_requests.map(r => r.id);
+
+    watch(props.ajax_requests, (old_v, new_v) => {
+        for (const request of new_v) {
+            if (seen_request_ids.indexOf(request.id) === -1) {
+                seen_request_ids.push(request.id);
+                setTimeout(() => {
+                    // Need this timeout because this watcher is called before the DOM is updated
+                    const row = $(`tr[data-request-id="${request.id}"]`);
+                    row.css('background-color', '#FFFF7B80');
+                    setTimeout(() => {
+                        row.css('background-color', 'transparent');
+                    }, 2000);
+                }, 10);
+            }
+        }
+    });
+
+    function expandRequestURL(e) {
+        const btn = $(e.currentTarget);
+        const td = btn.closest('td');
+        td.text(td.attr('title'));
+        btn.hide();
+    }
+
+    function selectRow(e) {
+        current_request_id.value = $(e.currentTarget).attr('data-request-id');
+        $(e.currentTarget).addClass('table-active').siblings().removeClass('table-active');
+    }
+
+    const active_subwidget = ref('request_summary');
+    const active_subwidget_component = computed(() => {
+        if (active_subwidget.value === null) {
+            return null;
+        }
+        const widget = props.widgets.find(w => w.id === active_subwidget.value);
+        if (widget === undefined) {
+            return null;
+        }
+        return widget.component_registered_name;
+    });
+    function switchSubwidget(widget_id) {
+        active_subwidget.value = widget_id;
+    }
+</script>
+
+<template>
+    <div ref="content_area" @mousemove="onMouseMove($event)" @mouseup="onMouseUp()">
+        <div class="request-timeline" v-if="show_timeline"></div>
+        <div class="d-flex flex-row h-100 split-panel-h">
+            <div class="left-panel" :style="`flex-basis: ${left_panel_flexbasis}`">
```

Comment:
Vue should allow us to use variables in the scoped style tag like `v-bind(left_panel_flexbasis)`. However, I've not been able to get this to work completely. It would replace the `v-bind` with a a reference to a CSS property, but it never initialized that property on the component element.

That is why there is an inline style here.

Reply:
`v-bind` is working properly now.